user_mail_common/res_users.py

```diff
@@ -114,34 +122,35 @@ def _remote_id(self):
 
     USER_MAIL_FIELDS = ['name', 'login',
                         'dovecot_password',
-                        'forward_active','forward_address','forward_cp',
-                        'postfix_active','postfix_alias_ids','postfix_mail',
+                        'forward_active', 'forward_address', 'forward_cp',
+                        'postfix_active', 'postfix_alias_ids', 'postfix_mail',
                         'quota',
-                        'spam_active','spam_killevel','spam_tag2','spam_tag',
-                        'vacation_subject','vacation_active','vacation_from',
-                        'vacation_to','vacation_forward','vacation_text',
+                        'spam_active', 'spam_killevel', 'spam_tag2', 'spam_tag',
+                        'vacation_subject', 'vacation_active', 'vacation_from',
+                        'vacation_to', 'vacation_forward', 'vacation_text',
                         'virus_active',
                         ]
     
-    @api.one
     @api.depends('company_id.domain', 'login')
     def _email(self):
-        if self.postfix_active and self.login:
-            email_re = re.compile(r"^[^@]+@[^@]+\.[^@]+$", re.VERBOSE)
-            if not email_re.match(self.login):  # login is not an email address
-                self.postfix_mail = '%s@%s' % (self.login, self.domain)
-            elif email_re.match(self.login):    # login is an (external) email address, use only left part
-                self.postfix_mail = '%s@%s' % (email_re.match(self.login).groups()[0],self.company_id.domain)
-    
-    @api.one
+        for this in self:
+            if this.postfix_active and this.login:
+                email_re = re.compile(r"""^([a-zA-Z0-9._%+-]+)@([a-zA-Z0-9.-]+\.[a-zA-Z]{2,6})$""", re.VERBOSE)
+                if not email_re.match(this.login):  # login is not an email address
+                    this.postfix_mail = '%s@%s' % (this.login, this.domain)
+                elif email_re.match(this.login):    # login is an (external) email address, use only left part
+                    this.postfix_mail = '%s@%s' % (email_re.match(this.login).groups()[0], this.company_id.domain)
+
```

Comment:
There is no else clause, can an else ever happen and what do we do then?

Reply:
I don't think there will be need for that, it is either a valid email or not